src/sp/it/pl/audio/Item.kt

```diff
@@ -24,9 +24,9 @@ abstract class Item {
 
     /**
      * @return absolute file this item represents
-     * @throws RuntimeException if this item is not file based
+     * @throws IllegalArgumentException if this item is not file based
      */
-    open fun getFile() = uri.toFileOrNull()!!.absoluteFile!!
+    open fun getFile() = File(uri).absoluteFile!!
```

Comment:
This obscures that there is a possibility of exception occurring unless the reader specifically knows the specifics of new File(URI). Javadoc states this, but that is for the user of the API, not reader of this code. Pls use the old version, it makes it absolutely clear that NullPointerException is the only possible error outcome.

Reply:
The Exception occuring is specified in the KDoc. But you swallow the exception to then spit out a Nullpointerexception, thus the original Exception message, which might be helpful, is lost.

Reply:
In a program that calls this method in allowable conditions, such case will never happen. Exception or any information is unnecessary. You wouldnt argue against the String.toIntOrNull() method either - it also swallows exception. There are contexts where such method is problematic, for the behavior you describe, this is not such a case.

Reply:
my issue was not the swallowing of the exception in itself, but swallowing it to only spill out another unrelated Exception afterwards

Reply:
any of this is irrelevant anyway, this is merely an assert we - if the developer calls the method when he should not, the program will crash. In the ideal scenario, there would be a throwIfNotFileBased() called somewhere, but this will be refactored to return null in such case anyway.

Reply:
How was this resolved?